core/rust/qdbr/src/parquet_write/schema.rs

```diff
@@ -329,15 +331,29 @@ pub struct Partition {
     pub columns: Vec<Column>,
 }
 
-pub fn to_parquet_schema(partition: &Partition) -> ParquetResult<SchemaDescriptor> {
+pub fn to_parquet_schema(
+    partition: &Partition,
+) -> ParquetResult<(SchemaDescriptor, Vec<KeyValue>)> {
     let parquet_types = partition
         .columns
         .iter()
         .map(|c| column_type_to_parquet_type(c.id, c.name, c.data_type))
         .collect::<ParquetResult<Vec<_>>>()?;
-    Ok(SchemaDescriptor::new(
-        partition.table.clone(),
-        parquet_types,
+
+    let additinal_keyvals = partition
+        .columns
+        .iter()
+        .map(|c| {
+            KeyValue::new(
+                QDB_TYPE_META_PREFIX.to_string() + c.name,
+                c.full_column_type.to_string(),
```

Comment:
why not update ColumnType to be "full". update try_from(v: i32) and ask ChatGPT to generate to_string/from_string type safe conversion?

Reply:
Full column type will include some column bit flags, like geohashes do. Maybe we do it later on but we rely on may places on those int values to be stable